dataactbroker/handlers/accountHandler.py

```diff
@@ -430,39 +423,67 @@ def email_users(self, system_email, session):
         return JsonResponse.create(StatusCode.OK, {"message": "Emails successfully sent"})
 
 
-def grant_superuser(user):
-    user.cgac_code = 'SYS'
-    user.website_admin = True
-    user.permission_type_id = PERMISSION_TYPE_DICT['writer']
-
-
-def grant_highest_permission(user, group_list, cgac_group_list):
-    """Find the highest permission within the provided cgac_group; set that as
-    the user's permission_type_id"""
-    sess = GlobalDB.db().session
-    user.website_admin = False
-    if not cgac_group_list:
-        user.cgac_code = None
-        user.permission_type_id = None
-        return
-
-    cgac_group = cgac_group_list[0]
-    permission_group = [g for g in group_list
-                        if g.startswith(cgac_group + "-PERM_")]
-    # Check if a user has been placed in a specific group. If not, deny access
-    if not permission_group:
-        user.permission_type_id = None
+def perms_to_affiliations(perms):
+    """Convert a list of perms from MAX to a list of UserAffiliations. Filter
+    out and log any malformed perms"""
+    available_codes = {
+        cgac.cgac_code:cgac
+        for cgac in GlobalDB.db().session.query(CGAC)
+    }
+    for perm in perms:
+        components = perm.split('-PERM_')
+        if len(components) != 2:
+            logger.warning('Malformed permission: %s', perm)
+            continue
+
+        cgac_code, perm_level = components
+        perm_level = perm_level.lower()
+        if cgac_code not in available_codes or perm_level not in 'rws':
+            logger.warning('Malformed permission: %s', perm)
+            continue
+
+        yield UserAffiliation(
+            cgac=available_codes[cgac_code],
+            permission_type_id=PERMISSION_SHORT_DICT[perm_level]
+        )
+
+
+def best_affiliation(affiliations):
+    """If a user has multiple permissions for a single agency, select the
+    best"""
+    by_agency = {}
+    affiliations = sorted(affiliations, key=attrgetter('permission_type_id'))
+    for affiliation in affiliations:
+        by_agency[affiliation.cgac] = affiliation
+    return by_agency.values()
+
+
+def set_max_perms(user, max_group_list):
+    """Convert the user group lists present on MAX into a list of
+    UserAffiliations and/or website_admin status.
+
+    Permissions are encoded as a comma-separated list of
+    {parent-group}-CGAC_{cgac-code}-PERM_{one-of-R-W-S}
+    or
+    {parent-group}-CGAC_SYS to indicate website_admin"""
+    prefix = CONFIG_BROKER['parent_group'] + '-CGAC_'
+    perms = [group_name[len(prefix):]
+             for group_name in max_group_list.split(',')
+             if group_name.startswith(prefix)]
```

Comment:
ahhh, trimming, that's what's happening in the first line there. Not sure why my brain refused to understand that. Might be nice to have a comment if you don't mind.